src/providers/postgres/qgspostgresconn.cpp

```diff
@@ -2354,7 +2354,11 @@ void QgsPostgresConn::retrieveLayerTypes( QVector<QgsPostgresLayerProperty *> &l
       switch ( zmFlags )
       {
         case 1:
-          typeString.append( 'M' );
+          // Unlike Geometry(Z), GeometryM contains already M suffix.
+          // It's useless to add M suffix here,
+          // since it will add a new name and the geometry type name will be something like GeometryMM.
+          // see: https://github.com/qgis/QGIS/issues/55223
+          //typeString.append( 'M' );
```

Comment:
Could we avoid introducing commented-out code here? :)

Reply:
Since I wasn't sure about my fix, and it still seems like something that deserves an explanation, I preferred to add my comment and leave the code as a comment. Do you agree, if I leave my comment, but delete the "faulty" line?